src/circuit_builder.rs

```diff
@@ -10,111 +10,64 @@ use crate::commitment_scheme::kzg10::PublicParameters;
-    pub const fn serialized_size() -> usize {
-        33usize
-    }
-
-    /// Returns the byte-representation of a [`PublicInput`].
-    /// Note that the underlying variants of this enum have different
-    /// sizes on it's byte-representation. Therefore, we need to return
-    /// the biggest one to set it as the default one.
-    pub fn to_bytes(&self) -> [u8; Self::serialized_size()] {
-        let mut bytes = [0u8; Self::serialized_size()];
-        match self {
-            Self::BlsScalar(scalar, _) => {
-                bytes[0] = BLS_SCALAR;
-                bytes[1..33].copy_from_slice(&scalar.to_bytes());
-                bytes
-            }
-            Self::JubJubScalar(scalar, _) => {
-                bytes[0] = JUBJUB_SCALAR;
-                bytes[1..33].copy_from_slice(&scalar.to_bytes());
-                bytes
-            }
-            Self::AffinePoint(point, _, _) => {
-                bytes[0] = JUBJUB_AFFINE;
-                bytes[1..Self::serialized_size()].copy_from_slice(&point.to_bytes());
-                bytes
-            }
-        }
+impl From<BlsScalar> for PublicInputValue {
+    fn from(scalar: BlsScalar) -> Self {
+        Self(vec![scalar])
     }
+}
 
-    /// Generate a [`PublicInput`] structure from it's byte representation.
-    pub fn from_bytes(bytes: &[u8]) -> Result<Self, Error> {
-        if bytes.len() < Self::serialized_size() {
-            return Err(Error::InvalidPublicInputBytes);
-        } else {
-            let mut array_bytes = [0u8; 32];
-            array_bytes.copy_from_slice(&bytes[1..Self::serialized_size()]);
-            match bytes[0] {
-                BLS_SCALAR => BlsScalar::from_bytes(&array_bytes)
-                    .map(|s| Self::BlsScalar(s, 0))
-                    .map_err(|_| Error::InvalidPublicInputBytes),
-
-                JUBJUB_SCALAR => JubJubScalar::from_bytes(&array_bytes)
-                    .map(|s| Self::JubJubScalar(s, 0))
-                    .map_err(|_| Error::InvalidPublicInputBytes),
-
-                JUBJUB_AFFINE => JubJubAffine::from_bytes(&array_bytes)
-                    .map(|s| Self::AffinePoint(s, 0, 0))
-                    .map_err(|_| Error::InvalidPublicInputBytes),
-
-                _ => unreachable!(),
-            }
-        }
+impl From<JubJubScalar> for PublicInputValue {
+    fn from(scalar: JubJubScalar) -> Self {
+        Self(vec![scalar.into()])
     }
+}
 
-    /// Returns the positions that of a PublicInput struct
-    fn pos(&self) -> [usize; 2] {
-        match self {
-            PublicInput::BlsScalar(_, pos) => [*pos, 0],
-            PublicInput::JubJubScalar(_, pos) => [*pos, 0],
-            PublicInput::AffinePoint(_, pos_x, pos_y) => [*pos_x, *pos_y],
-        }
+impl From<JubJubAffine> for PublicInputValue {
+    fn from(point: JubJubAffine) -> Self {
+        Self(vec![point.get_x(), point.get_y()])
     }
 }
 
+type PublicInputPositions = Vec<usize>;
```

Comment:
I would avoid the type alias here, for the following reasons:

1. It's actually a way longer than `Vec<usize>` to type
2. It's only local, and we're using in three places only.
3. Two of them are arguments, so there is already the argument's name to indicate what they're used for, so it's a kind of redundant.
4. In this two places, you're actually passing a reference. So it's like you're passing `&Vec<usize>` and in a regular code you would never dream doing so, since you're basically using two references when you could use just one.

So I would remove that, and add the proper type in the three places mentioned (see the other comments below).